experimental/dds/tree2/src/test/shared-tree-core/editManagerTestUtils.ts

```diff
@@ -42,34 +42,36 @@ export function editManagerFactory(options: {
 }
 
 export function rebaseLocalEditsOverTrunkEdits(
-	nbLocal: number,
-	nbTrunk: number,
+	localEditCount: number,
+	trunkEditCount: number,
 	rebaser: TestChangeRebaser,
-): void {
+	defer: boolean = false,
+): void | (() => void) {
 	const manager = editManagerFactory({ rebaser }).manager;
-	for (let iChange = 0; iChange < nbLocal; iChange++) {
+	for (let iChange = 0; iChange < localEditCount; iChange++) {
 		manager.localBranch.apply(TestChange.emptyChange, mintRevisionTag());
 	}
-	for (let iChange = 0; iChange < nbTrunk; iChange++) {
-		manager.addSequencedChange(
-			{
-				change: TestChange.emptyChange,
-				revision: mintRevisionTag(),
-				sessionId: "trunk",
-			},
-			brand(iChange + 1),
-			brand(iChange),
-		);
-	}
+	const trunkEdits = makeArray(trunkEditCount, () => ({
+		change: TestChange.emptyChange,
+		revision: mintRevisionTag(),
+		sessionId: "trunk",
+	}));
+	const run = () => {
+		for (let iChange = 0; iChange < trunkEditCount; iChange++) {
+			manager.addSequencedChange(trunkEdits[iChange], brand(iChange + 1), brand(iChange));
+		}
+	};
+	return defer ? run : run();
 }
 
 export function rebasePeerEditsOverTrunkEdits(
-	nbPeer: number,
-	nbTrunk: number,
+	peerEditCount: number,
+	trunkEditCount: number,
 	rebaser: TestChangeRebaser,
-): void {
+	defer: boolean = false,
+): void | (() => void) {
```

Comment:
Interesting pattern! This pattern [works well with overloads so that you don't need to check or assert the return type](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMAFAEwKbAwJwPwBcyAhgDYDOGAlEQG5wxoDcAsAFCiSwLLpa5EoOENSIoqiALwA+RPUasO4aPCSpM2HFJIVRchmkQAfRCnFTZ8tBIDe7RA8Qxgpjblv3HXnBiggcSOYyiDYAvopeoexRbOwQCORQiMTaqFRMXg4A9FnJTuT6jHEJSQBGqSjAZJTpiDmI5TAFVsVgiYgQFUIitXW5nU2mEsFWQA) for hardcoded use cases.